### PR TITLE
Feature presidecms 576 asset manager issue

### PR DESCRIPTION
**Pull request by tommyw1314**
fix error on duplicated file exist in target folder

### CODE REVIEW

**Review by DominicWatson — CHANGES_REQUESTED** [review]
_Deleting this review to try again properly_ (nothing to see here)

**Review by DominicWatson — APPROVED** [review]

File: system/services/assetManager/AssetManagerService.cfc [hunk 1]
```diff
@@ -210,13 +211,16 @@ component displayName="AssetManager Service" {
 		  required string  type
 		, required string  size
 		, required string  folderId
+		,          string  assetId
```
Comment (DominicWatson) [inline system/services/assetManager/AssetManagerService.cfc]: This argument should either be required or have a default value (""). I think it should probably be required.

File: system/services/assetManager/AssetManagerService.cfc [hunk 2]
```diff
@@ -210,13 +211,16 @@ component displayName="AssetManager Service" {
...
 		,          string  currentFolderId = ""
 		,          boolean throwIfNot   = false
 		,          struct  restrictions = getFolderRestrictions( arguments.folderId )
 	) {
 		var typeDisallowed = restrictions.allowedExtensions.len() && !ListFindNoCase( restrictions.allowedExtensions, "." & arguments.type );
 		var sizeInMb       = arguments.size / 1048576;
 		var tooBig         = restrictions.maxFileSize && sizeInMb > restrictions.maxFileSize;
+		var fileExist      = _getAssetDao().dataExists( filter = { id = arguments.assetId } );
```
Comment (DominicWatson) [inline system/services/assetManager/AssetManagerService.cfc]: What we should be checking for here is that _another_ asset does not already exist in the new folder that has the same _name_ as the current asset.

If we just check for the existance of the asset, then we will never be able to move assets between folders.

**Review by DominicWatson — CHANGES_REQUESTED** [review]
I believe there are some issues around the logic used to check uniqueness of assets within a folder. Please see review comments.

**Comment by tommyw1314** [comment]
i have removed the assetId and do checking based on the asset_folder & title.

**Comment by DominicWatson** [comment]
Great, thanks @tommyw1314. Merging into the upcoming release.

**Pull request merged**